File: design-patterns/go/visitor/hosting_cost/serverless_service.go
// serverless_service.go

package main

type ServerlessService struct {
	hourlyPrice float64
	totalHours  int
}

func NewServerlessService(totalHours int) (serverlessService *ServerlessService) {
	serverlessService = &ServerlessService{}
	serverlessService.hourlyPrice = 0.32
	serverlessService.totalHours = totalHours
	return
}

func (serverlessService *ServerlessService) GetHourlyPrice() float64 {
	return serverlessService.hourlyPrice
}

func (serverlessService *ServerlessService) GetTotalHours() int {
	return serverlessService.totalHours
}

func (serverlessService *ServerlessService) Accept(hostingCalculatorVisitor HostingCalculatorVisitor) float64 {
	return hostingCalculatorVisitor.ServerlessVisit(serverlessService)
}

File: design-patterns/php/visitor/hostingcost/ServerlessService.php
<?php
// ServerlessService.php

namespace BigBoxCode\DesignPattern\Visitor\HostingCost;

class ServerlessService implements Service {
    private const HOURLY_PRICE = 0.32;
    private int $totalHours;

    public function __construct(int $totalHours) {
        $this->totalHours = $totalHours;
    }

    public function getHourlyPrice(): float {
        return self::HOURLY_PRICE;
    }

    public function getTotalHours(): int {
        return $this->totalHours;
    }

    public function accept(HostingCalculatorVisitor $hostingCalculatorVisitor): float {
        return $hostingCalculatorVisitor->visitServerlessService($this);
    }
}